src/Agent.Worker/TestResults/TestCaseSubResultData.cs

```diff
@@ -0,0 +1,49 @@
+using System;
+using System.Collections.Generic;
+using Microsoft.TeamFoundation.TestManagement.WebApi;
+
+namespace Microsoft.VisualStudio.Services.Agent.Worker.TestResults
+{
+    public class TestCaseSubResultData
```

Comment:
You can move some of the common fields to base class and inherit from both TestCaseResultData and TestCaseSubResultData.

Reply:
This is how we are doing this in our Publish task, Therefore, keeping it same. Moreover, there aren't many common fields. So, making a class for common fields wont reduce the number of lines of code significantly.